# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1546635667 -3600
#      Fri Jan 04 22:01:07 2019 +0100
# Branch stable
# Node ID 93c4fabb2ead1be8654de1aa119cdf7d96f4406e
# Parent  93c7fc75b1039e52d16b89ecb2820706bcb94b77
only reset floating geometry of dock widget when not visible (bug #53207)

* octave-dock-widget.cc: update mail address in copyright notice;
  (handle_settings): compute default floating size based on geometry of
  screen of the main window, test if restored geometry is on a screen, if
  not, use default size

diff --git a/libgui/src/octave-dock-widget.cc b/libgui/src/octave-dock-widget.cc
--- a/libgui/src/octave-dock-widget.cc
+++ b/libgui/src/octave-dock-widget.cc
@@ -1,12 +1,12 @@
 /*
 
 Copyright (C) 2012-2019 Richard Crozier
-Copyright (C) 2013-2019 Torsten <ttl@justmail.de>
+Copyright (C) 2013-2019 Torsten <mttl@mailbox.org>
 
 This file is part of Octave.
 
 Octave is free software: you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License, or
 (at your option) any later version.
 
@@ -438,26 +438,29 @@ namespace octave
       m_icon_color = "";
 
     bcola.getRgb (&r, &g, &b);
     if (r+g+b < 400)
       m_icon_color_active = "-light";
     else
       m_icon_color_active = "";
 
-    QRect available_size = QApplication::desktop ()->availableGeometry ();
+    QRect available_size = QApplication::desktop ()->availableGeometry (m_parent);
     int x1, y1, x2, y2;
     available_size.getCoords (&x1, &y1, &x2, &y2);
     QRect default_size = QRect (x1, y1, x2/3, y2/2);
 
     m_recent_float_geom = settings->value ("DockWidgets/" + objectName ()
                                            + "_floating_geometry",
                                            default_size).toRect ();
 
-    if (! available_size.contains (m_recent_float_geom, false))
+    QWidget dummy;
+    dummy.setGeometry (m_recent_float_geom);
+
+    if (QApplication::desktop ()->screenNumber (&dummy) == -1)
       m_recent_float_geom = default_size;
 
     m_recent_dock_geom = settings->value ("DockWidgets/" + objectName (),
                                           QByteArray ()).toByteArray ();
 
     notice_settings (settings);  // call individual handler
 
     set_style (false);
